producer/src/main/java/org/aya/producer/AyaProducer.java

```diff
@@ -293,19 +297,31 @@ private void pragma(GenericNode<?> node, Decl decl) {
     node.childrenOfType(PRAGMA).forEach(pragma -> {
       var nameToken = pragma.child(ID);
       var name = nameToken.tokenText().toString();
+      var namePos = sourcePosOf(nameToken);
       var argsNode = pragma.peekChild(COMMA_SEP);
       var args = argsNode != null ? argsNode.childrenOfType(ID) : SeqView.<GenericNode<?>>empty();
+
       switch (name) {
-        case "suppress" -> args.forEach(arg -> {
-          for (var suppress : Suppress.values()) {
-            if (arg.tokenText().contentEquals(suppress.name())) {
-              decl.suppresses.add(suppress);
-              return;
+        case Constants.PRAGMA_SUPPRESS -> {
+          // TODO: use MutableEnumSet
+          MutableSet<WithPos<Suppress>> sups = MutableSet.create();
```

Comment:
I think I used a set because I want each suppress to be unique, but now with a source pos the same warning can be suppresses many times in the same set.
I suggest using a list instead

Reply:
I am not sure if we really need the source pos.

Reply:
I think we do, it's needed for faithful highlight of them